projects/shared/src/lib/interceptor/spinner.interceptor.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2021 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest, HttpResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Observable, of } from 'rxjs';
+import { catchError, tap } from 'rxjs/operators';
+
+@Injectable()
+export class SpinnerInterceptor implements HttpInterceptor {
+
+  /** Request count */
+  private _requestCount = 0;
+
+  /**
+   * Constructor
+   * @param _spinnerService - NgxSpinnerService
+   */
+  constructor(private _spinnerService: NgxSpinnerService) {}
+
+  /**
+   * Intercept
+   * @param request - HttpRequest
+   * @param next - HttpHandler
+   * @return Observable
+   */
+  intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+    this._spinnerService.show();
+    this._requestCount++;
+    return next.handle(request).pipe(
+      tap(event => {
+        if (event instanceof HttpResponse) {
+          this._requestCount--;
+          if (this._requestCount === 0) {
+            this._spinnerService.hide();
+          }
+        }
+      }),
+      catchError((err: any) => {
+        this._spinnerService.hide();
+        return of(err);
+      })
+    );
+  }
```

Comment:
As I understand, the spinner will be shown each time an HTTP call is pending.... despite if this call is manage as an asynchronous call. So my conclusion is that if a patron has 20 pendings loans, he needs to wait that all the HTTP calls are ended before to do something on the page ! 

Spinner is a good thing to show that application is working/waiting data/... but use a spinner to block application until all data are available is not possible for a 21 century application.

You already mentioned this module https://aitboudad.github.io/ngx-loading-bar/ (same developper than ngx-formly). Maybe it's time to test/use it.

Reply:
Globally agree with @zannkukai. It's not a good solution to show the blocking spinner for each HTTP calls. And I'm not sure that `ngx-loading-bar` would be better. We have to consider the blocking and non-blocking operations for each case. An "automatic" solution cannot be used.

Reply:
I agree with @zannkukai. It's not correct for a "modern application"
Another observation : Have you tried the responsive interface of the new profil ? :-\

Reply:
I tested the new "ngx-loading-bar" module on the web component. It works perfectly and it doesn't block the page.

Reply:
Remove spinner interceptor